### ISSUE TITLE
CSV Table Viewer / Editor

### ISSUE THREAD

**Issue by JosephTLyons** [zed-industries/extensions#476]

**Comment by Eyal4K** [comment]
very much needed!

**Comment by julienlag** [comment]
I strongly support this!

**Comment by nwhistler** [comment]
It's the one feature I am missing from Zed

**Comment by mwfjord** [comment]
Agreed! I really enjoy data wranglers workflow and its integration with copilot.  https://marketplace.visualstudio.com/items?itemName=ms-toolsai.datawrangler
Would be really neat to see something like this become available!

**Comment by pintert3** [comment]
Coming here after enjoying the power of csvlens. I reckon it could be a good starting point

**Comment by swarajban** [comment]
yea we need this

**Comment by mwfjord** [comment]
I would also add that the core feature that makes data wrangler so powerful in vscode, is that it connects seamlessly with the debugger. So, when working with any type of tabular data format, having the ability to break, dig into the dataframe, and view the state of all rows and columns makes it incredibly good for data science and data engineering pipelines. I'm currently doing this by running visidata with neovim. I think if you just offer the option to set visidata, for example, as a default dataframe inspector in the debugger, you will have a very powerful starting point without too much implementation.

**Comment by FangyangJz** [comment]
Datawrangler,  great extension !

**Comment by johanidler** [comment]
Would be very helpfull!

**Comment by VBoudreau55** [comment]
I would love to see something like Data wrangler. Possibly instead of tying closely to Pandas, it could use something like [Polars](https://pola.rs/) for the dataframe engine

**Comment by GibreelAbdullah** [comment]
We need this

**Comment by coxjc** [comment]
PLEASE

**Comment by kushvinth** [comment]
Is anyone working on this, or is there any plugin/extension that solves this issue?

**Comment by KanTakahiro** [comment]
We can use CMD+Shift+V to render Markdown file. I think there can be a same CMD+Shift+V shortcut for viewing CSV file as a table.

**Comment by kushvinth** [comment]
The format needs to be .csv

**Comment by JosephTLyons** [comment]
CSV table previews have shipped in [Zed Stable v1.17.2](https://zed.dev/releases/stable)!

The feature only covers previewing at this time, so I'm leaving the issue open.

Thanks to [@HalavicH](https://github.com/HalavicH) for the implementation!

**Comment by kushvinth** [comment]
Massive @JosephTLyons

**Comment by b4d** [comment]
I would love to see the edit CSV table view!

Is there any way to make ';' separated CSV display properly in the table preview?
